lib/features/family/features/history/history_cubit/history_cubit.dart

```diff
@@ -12,22 +12,35 @@ class HistoryCubit extends Cubit<HistoryState> {
   HistoryCubit(this.historyRepo) : super(const HistoryState());
   final HistoryRepository historyRepo;
 
-  FutureOr<void> fetchHistory(String childId) async {
+  FutureOr<void> fetchHistory(
+    String childId, {
+    bool fromBeginning = false,
+  }) async {
+    // Make sure that it's not loading already
+    if (state.status == HistoryStatus.loading) return;
     emit(state.copyWith(status: HistoryStatus.loading));
 
+    if (fromBeginning) emit(state.copyWith(history: [], pageNr: 1));
+
     try {
-      final donationHistory = await historyRepo.fetchHistory(
+      final donationHistoryFuture = historyRepo.fetchHistory(
         childId: childId,
         pageNumber: state.pageNr,
         type: HistoryTypes.donation,
       );
 
-      final allowanceHistory = await historyRepo.fetchHistory(
+      final allowanceHistoryFuture = historyRepo.fetchHistory(
         childId: childId,
         pageNumber: state.pageNr,
         type: HistoryTypes.allowance,
       );
 
+      // In this way the second future will already start fetching
+      // while waiting for the first one to finish
+      // This will make the fetching faster
+      final donationHistory = await donationHistoryFuture;
```

Comment:
I don't think it does what you say it does...

![Scherm­afbeelding 2024-06-12 om 09 51 52](https://github.com/givtnl/givt-app/assets/2879338/81aaef3d-eb25-432b-9c8d-7b4a399fd8dd)

If you want to fetch the calls in parallel I would recommend the Future.wait([]) function:
https://api.dart.dev/stable/3.4.3/dart-async/Future/wait.html

so:

`var responses = await Future.wait([donationHistoryFuture, allowanceHistoryFuture])`

Reply:
You can have futures without awaiting them :) they also will get executed immediately.

Reply:
so the execution process will start already before the first await

Reply:
I think the future.wait is only needed if you need to be sure at that point that it has both results. But in our case its fine if it doesnt have the second one yet

Reply:
Good point, indeed nothing needs to be changed then!